metagraph/src/graph/alignment/dbg_aligner.hpp

```diff
@@ -50,11 +52,15 @@ class ISeedAndExtendAligner : public IDBGAligner {
     const DBGAlignerConfig& get_config() const { return config_; }
 
   protected:
-    typedef AlignmentAggregator<AlignmentCompare> Aggregator;
+    typedef std::vector<std::tuple<std::shared_ptr<ISeeder>, std::vector<node_index>,
+                                   std::shared_ptr<ISeeder>, std::vector<node_index>>> BatchSeeders;
     const DeBruijnGraph &graph_;
+    DBGAlignerConfig config_;
 
     virtual std::shared_ptr<IExtender>
-    build_extender(std::string_view query, const Aggregator &aggregator) const = 0;
+    build_extender(std::string_view query, const DBGAlignerConfig &config) const = 0;
```

Comment:
I think it makes sense to pass Seeder and Extender as template parameters to this class, so you don't go via 
interfaces ISeeder and IExtender but call them directly.

Reply:
If it was designed this way, then one would have to either move the implementations to the hpp file, or instantiate this class with every combination of Seeder and Extender types. So, I thought doing it this way instead would be a good compromise.